### ISSUE TITLE
Support custom options(extensions) for MessageOptions/FieldOptions/MethodOptions/...

### ISSUE THREAD

**Issue by c98** [andrewhickman/prost-reflect#21]
I have seen this closed issue #8 , and It seems there is no custom options support right now.
I came from c++, the [impl](https://github.com/protocolbuffers/protobuf/blob/main/src/google/protobuf/descriptor.pb.h#L5150) contains a _extensions_ for custom options support. 

Our application use the custom options heavily, so can we support this feature? Thanks.

**Comment by andrewhickman** [comment]
This is something I'd like to implement as well because extension options are currently the biggest gap in my [protox](https://docs.rs/protox/latest/protox/) crate. I hope to get to this soon.

The main issue is that we use `prost_reflect::FileDescriptorSet` internally which doesn't include any custom options

Its possible (though extremely unergonomic) to get these options using the current API - see https://github.com/tokio-rs/prost/issues/635#issuecomment-1179566865. Perhaps we could add some convenience APIs to make it simpler

**Comment by c98** [comment]
Thanks for your reply, i'm looking forward to seeing it soon ("the convenience APIs")
come on ;)

**Comment by andrewhickman** [comment]
I've released an alpha version with better support for custom options:
https://docs.rs/prost-reflect/0.10.0-alpha.1/prost_reflect/struct.MessageDescriptor.html#method.options
Extension options should now be preserved when creating a `DescriptorPool` from bytes, and they can be queried with a new `options()` method on each descriptor. Usage would be something like

```rust
let pool = DescriptorPool::decode(...);

let message = pool.get_message_by_name("package.Message");
let extension_option = pool.get_extension_by_name("package.custom_message_option");

let message_options = message.options();
assert!(message.has_extension(&extension_option));
assert_eq!(message.options.get_extension(&extension_option).as_str(), Some("value"));
```
where the proto file looks like
```proto
import 'google/protobuf/descriptor.proto';

package package;

extend google.protobuf.MessageOptions {
    optional string custom_message_option = 1001;
}

message Message {
    option (custom_message_option) = "value";
}
```

**Comment by andrewhickman** [comment]
Implemented in version 0.10.0
